packages/itwin/reports-config-widget/src/widget/components/Reports.tsx

```diff
@@ -107,6 +114,28 @@ export const Reports = () => {
     [reports, searchValue]
   );
 
+  useEffect(() => {
+    if (jobStarted) {
+      const interval = setInterval(async () => {
+        const states = await bulkExtractor.getStates(reports.map(r => r.id));
+        if (Array.from(states.values())
+          .filter(x => x === ExtractionStates.Succeeded ||
+            x === ExtractionStates.Failed ||
+            x === ExtractionStates.None)
+          .length === states.size) {
+          setJobStarted(false);
+          bulkExtractor.clearJobs();
+        }
+        setExtractionStates(states);
+      }, 5000)
+      return () => clearInterval(interval);
+    }
+    setTimeout(() => {
+      setExtractionStates(new Map<string, ExtractionStates>());
+    }, 5000);
```

Comment:
This setTimeout can be moved into its own useEffect with its own clearTimeout cleanup.

Reply:
`setTimeout` has been moved into new useEffect